src/App.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react'
+import './client/App.css'
+import Header from './Header'
+import Footer from './Footer'
+import AllArticles from './AllArticles'
+import SeparatePost from './SeparatePost'
+import { BrowserRouter as Router, Route } from 'react-router-dom'
+import marked from 'marked'
+
+const SERVER_ROOT = 'https://cmsblog-aaa94.firebaseio.com'
+
+/* global fetch */
+
+class App extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      markdownList: [],
+      config: [],
+      contact: [],
+      titles: []
+    }
+  }
+  componentWillMount () {
+    fetch(`${SERVER_ROOT}/articles.json`)
+      .then(response => {
+        console.log('It worked!')
+        console.log(response)
+        return response.json()
+      })
+      .then(text => {
+        console.log('It worked1')
+        console.log(text)
+        const newMarkdownList = this.state.markdownList.map((item) => Object.assign({}, item))
+        for (const x in text) {
+          console.log('________', x, '__________')
+          newMarkdownList.push({content: marked(text[x].data), path: x})
+          this.setState({
+            markdownList: newMarkdownList
+          })
+        }
```

Comment:
This code is problematic for two reasons: (1) there is never any reason to use `for`– there are at least two other ways of writing this code that are cleaner and easier to maintain. (2) you should not do multiple `setState()` operations as part of `componentWillMount()` You should only do a single `setState()` operation. Therefore I would suggest using a `forEach()` as follows:

```javascript
const newMarkdownList = this.state.markdownList.map((item) => Object.assign({}, item))
text.forEach(x => newMarkdownList.push(x))
this.setState({
  markdownList: newMarkdownList
})
```